### PR TITLE
feat: adding not enough RIF balance before troughing transaction

### PR DESCRIPTION
**Pull request by annipi**
[REM-31](https://rsklabs.atlassian.net/browse/REM-31)

The user should not be able to start domain purchasing operation if they do not have enough RIF balance to save gas.

Screenshot 2024-02-22 at 11 15 00 AM

### CODE REVIEW

File: src/app/tabs/registrar/actions.js [hunk 1]
```diff
@@ -103,3 +104,12 @@ export const toggleSetupAddr = setupAddr => ({
 export const closeErrorMessage = () => ({
   type: CLOSE_REGISTRATION_ERROR,
 });
+
+export const requestHasEnoughRif = () => ({
+  type: REQUEST_HAS_ENOUGH_RIF,
+});
+
+export const errorHasEnoughRIF = message => ({
```
Comment (DinizSa) [inline src/app/tabs/registrar/actions.js:112]: should 'errorHasEnoughRIF' be renamed to 'errorNotEnoughRIF' to be more clear?

File: src/app/tabs/registrar/operations.js [hunk 2]
```diff
@@ -214,7 +217,13 @@ export const checkIfAlreadyCommitted = domain => async (dispatch) => {
   }
 };
 
-export const revealCommit = domain => async (dispatch) => {
+export const revealCommit = domain => async (dispatch, getState) => {
+  const { rifCost: cost } = getState().registrar;
+
+  const errorMessage = getLanguageString('not_enough_rif_balance');
```
Comment (DinizSa) [inline src/app/tabs/registrar/operations.js:223]: is there a way to avoid the 'not_enough_rif_balance' magic value? if there is no easy way is ok.

**Review by AndresQuijano — CHANGES_REQUESTED** [review]
I found two things:
-The message is always in English despite the lenguaje configuration.
-I found a bug:
1. Try to buy a domain with a wallet with 0 RIF.
2. It shows the error message.
3. Fund the same wallet with some RIFs.
4. Try again. 
5. The error message is shown anyway.

File: src/app/components/UserErrorComponent.js [hunk 3]
```diff
@@ -43,6 +44,8 @@ const UserErrorComponent = ({
         return strings.could_not_encode_address;
       case TRANSACTION_RECEIPT_FAILED:
         return strings.transaction_receipt_failed;
+      case ERROR_NOT_ENOUGH_RIF:
```
Comment (antomor) [inline src/app/components/UserErrorComponent.js:47]: NIT: @annipi I know you didn't introduce it, but what do you think if we refactor this with something like:
```
const errorMessages = {
    ERROR_SAME_VALUE: strings.same_value,
    ERROR_RESOLVE_NAME: strings.resolve_not_set,
    ADDRESS_ENCODING_ERROR: strings.could_not_encode_address,
    TRANSACTION_RECEIPT_FAILED: strings.transaction_receipt_failed,
    ERROR_NOT_ENOUGH_RIF: strings.not_enough_rif_balance,
  };

  const formatMessage = () => errorMessages[message] || truncateErrorMessage(message);
```

**Comment by antomor** [comment]
I've also tried to reproduce the error experienced by @AndresQuijano, but I couldn't

File: src/app/tabs/registrar/actions.js (hunk 1 above)
Comment (annipi) [reply src/app/tabs/registrar/actions.js]: done!

File: src/app/tabs/registrar/operations.js (hunk 2 above)
Comment (annipi) [reply src/app/tabs/registrar/operations.js:223]: Jummm `not_enough_rif_balance` is the key on the languages json files, I don't see other way of doing this.

File: src/app/components/UserErrorComponent.js (hunk 3 above)
Comment (annipi) [reply src/app/components/UserErrorComponent.js]: Ok, done!

File: src/app/tabs/registrar/operations.js (hunk 2 above)
Comment (annipi) [reply src/app/tabs/registrar/operations.js]: Finally I find a way to load the corresponding `errorMessage` without needing to use `getLanguageString` function.

**Comment by annipi** [comment]
> I found two things: -The message is always in English despite the lenguaje configuration. -I found a bug:
> 
> 1. Try to buy a domain with a wallet with 0 RIF.
> 2. It shows the error message.
> 3. Fund the same wallet with some RIFs.
> 4. Try again.
> 5. The error message is shown anyway.

1. I fixed the issue of the error message not changing when the language changed.
2. Apparently this behavior of still seeing an error if the error box not explicity closed(if the page gets reload this does happen), is consistent for all other error messages as well, we can create a ticket to address this, at least for the `develop` branch since this is an improvement.

File: src/app/components/UserErrorComponent.js (hunk 3 above)
Comment (annipi) [reply src/app/components/UserErrorComponent.js]: @antomor  I didn't initially understand the change, now i t it fixed, I removed the switch case and make `errorMessages` an object.

**Review by antomor — APPROVED** [review]
LGTM

**Review by AndresQuijano — APPROVED** [review]

**Pull request merged**